pkg/lib/operatorstatus/status.go

```diff
@@ -131,17 +140,24 @@ func MonitorClusterStatus(name string, syncCh chan error, stopCh <-chan struct{}
 			})
 			// we set the versions array when all the latest code is deployed and running - in this case,
 			// the sync method is responsible for guaranteeing that happens before it returns nil
-			if version := os.Getenv("RELEASE_VERSION"); len(version) > 0 {
+			if len(targetOperatorVersion) > 0 {
 				existing.Status.Versions = []configv1.OperandVersion{
 					{
 						Name:    "operator",
-						Version: version,
+						Version: targetOperatorVersion,
 					},
 					{
 						Name:    "operator-lifecycle-manager",
 						Version: olmversion.OLMVersion,
 					},
 				}
+				if targetOperatorVersion != previousOperatorVersion {
```

Comment:
Just to check my understanding - on cluster init, `previousOperatorVersion` _should_ be "0.0.1-snapshot" here, right?

Reply:
it should be, yes